app/models/api/messages/useq_wafer_io.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+# Generates warehouse messages describing a Ultima Genomics Wafer. While wafers are not
+# modeled directly in Sequencescape they can be approximated by a sequencing
+# Wafers are equivalent to a flowcell in other sequencing platforms hence much of the code is similar to FlowcellIo
+# {Batch}
+class Api::Messages::UseqWaferIo < Api::Base
+  self.includes = {
+    requests: [
+      {
+        target_asset: {
+          aliquots: [
+            :library,
+            { tag: :tag_group, sample: :uuid_object, study: :uuid_object, project: :uuid_object }
+          ]
+        }
+      },
+      :lab_events,
+      :batch_request,
+      :request_metadata
+    ]
+  }
+
+  # The following modules add methods onto the relevant models, which are used below in generation of the
+  # useq wafer MLWH message.
+  # Included in UltimaSequencingRequest model
+  module LaneExtensions
+    def self.included(base) # rubocop:todo Metrics/MethodLength
+      base.class_eval do
+        # Lot numbers for opentron and amp
+        def otr_carrier_lot_number
+          detect_descriptor('OTR carrier Lot #')
+        end
+
+        def otr_carrier_expiry
+          detect_descriptor('OTR carrier expiry')
+        end
+
+        def otr_reaction_mix_7_lot_number
+          detect_descriptor('Reaction Mix 7 Lot #')
+        end
+
+        def otr_reaction_mix_7_expiry
+          detect_descriptor('Reaction Mix 7 expiry')
+        end
+
+        def otr_nfw_lot_number
+          detect_descriptor('NFW Lot #')
+        end
+
+        def otr_nfw_expiry
+          detect_descriptor('NFW expiry')
+        end
+
+        def otr_oil_lot_number
+          detect_descriptor('Oil Lot #')
+        end
+
+        def otr_oil_expiry
+          detect_descriptor('Oil expiry')
+        end
+
+        def otr_pipette_carousel
+          detect_descriptor('Pipette carousel')
+        end
+
+        def otr_instrument_name
+          detect_descriptor('Opentrons Inst. Name')
+        end
+
+        def amp_assign_control_bead_tube
+          detect_descriptor('Assign Control Bead Tube')
+        end
+
+        def amp_instrument_name
+          detect_descriptor('UG AMP Inst. Name')
+        end
+      end
+    end
+  end
```

Comment:
note: matches the ultima descriptors config/default_records/descriptors/003_ultima_descriptors.wip.yml